### PR TITLE
Refactor tx status computer

### PR DESCRIPTION
**Pull request by valentincraciun99**
refactoring and nil checks  in transaction status computer

### CODE REVIEW

File: data/transaction/status.go
Comment (mateigraura) [inline data/transaction/status.go:39]: SelfShardID would be more explicit

File: data/transaction/status_test.go
Comment (mateigraura) [inline data/transaction/status_test.go:10]: please maintain the imports format

**Review by mateigraura — CHANGES_REQUESTED** [review]

File: data/transaction/interface.go
Comment (iulianpascalau) [inline data/transaction/interface.go:21]: Missing comment on exported item (please set your IDE to point out this)

File: data/transaction/status.go
Comment (iulianpascalau) [inline data/transaction/status.go:41]: Missing comment

File: data/transaction/status.go
Comment (iulianpascalau) [inline data/transaction/status.go:49]: Please reformat this as follows:
```GO
func (sc *StatusComputer) ComputeStatusWhenInStorageKnowingMiniblock(
	miniblockType block.Type,
	tx *ApiTransactionResult,
) TxStatus {
	isMiniblockFinalized := tx.NotarizedAtDestinationInMetaNonce > 0
...


```

File: data/transaction/status.go
Comment (iulianpascalau) [inline data/transaction/status.go:53]: This variable could have been avoided. `tx.DestinationShard` expresses the intent.

File: data/transaction/status.go
Comment (iulianpascalau) [inline data/transaction/status.go:55]: here too, `tx.Data` is pretty expressive

File: data/transaction/status.go
Comment (iulianpascalau) [inline data/transaction/status.go:73]: can remove the leading tab here

File: data/transaction/status.go
Comment (iulianpascalau) [inline data/transaction/status.go:75]: useless variable?

File: data/transaction/status.go
Comment (iulianpascalau) [inline data/transaction/status.go:105]: can remove the leading tab

File: data/transaction/status.go
Comment (iulianpascalau) [inline data/transaction/status.go:103]: check for nils for `tx`, `uint64ByteSliceConverter`, `store`?
Since the method is exported, the code can easily break if someone will pass nil on some of the parameters.

Why not passing the `uint64ByteSliceConverter` and `store` as arguments on the constructor?

File: data/transaction/status.go
Comment (iulianpascalau) [inline data/transaction/status.go:72]: check for nil?

File: data/transaction/status.go
Comment (iulianpascalau) [inline data/transaction/status.go:51]: check for nil?

File: data/transaction/status.go
Comment (iulianpascalau) [inline data/transaction/status.go:37]: why do we keep the StatusComputer exported?

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

**Review by iulianpascalau — CHANGES_REQUESTED** [review]
Also, please add description on the PR

File: data/transaction/status.go
Comment (valentincraciun99) [reply data/transaction/status.go:42]: ok, Done!

File: data/transaction/status_test.go
Comment (valentincraciun99) [reply data/transaction/status_test.go:11]: Done!

File: data/transaction/interface.go
Comment (valentincraciun99) [reply data/transaction/interface.go:20]: Done!

File: data/transaction/status.go
Comment (valentincraciun99) [reply data/transaction/status.go]: Done!

File: data/transaction/status.go
Comment (valentincraciun99) [reply data/transaction/status.go:60]: Done!

File: data/transaction/status.go
Comment (valentincraciun99) [reply data/transaction/status.go]: Done!

File: data/transaction/status.go
Comment (valentincraciun99) [reply data/transaction/status.go]: Done!

File: data/transaction/status.go
Comment (valentincraciun99) [reply data/transaction/status.go]: Done!

File: data/transaction/status_test.go
Comment (mateigraura) [inline data/transaction/status_test.go:17]: why don't you make use of the `NewStatusComputer` you added ?

File: data/transaction/status_test.go
Comment (mateigraura) [inline data/transaction/status_test.go:81]: same, why don't you use the constructor ?

Comment (valentincraciun99) [reply data/transaction/status_test.go:77]: Done!

File: data/transaction/status_test.go
Comment (valentincraciun99) [reply data/transaction/status_test.go:15]: Done!

File: data/transaction/status.go
Comment (valentincraciun99) [reply data/transaction/status.go]: Done!

File: data/transaction/status.go
Comment (valentincraciun99) [reply data/transaction/status.go]: Done!

File: data/transaction/status.go
Comment (valentincraciun99) [reply data/transaction/status.go:86]: Done!

File: data/transaction/status.go
Comment (valentincraciun99) [reply data/transaction/status.go]: Done!

File: node/blockAPI/metaBlock.go
Comment (miiu96) [inline node/blockAPI/metaBlock.go]: format

File: node/blockAPI/shardBlock.go
Comment (miiu96) [inline node/blockAPI/shardBlock.go]: fix format

File: data/transaction/status.go
Comment (valentincraciun99) [reply data/transaction/status.go]: Done!

File: data/transaction/status.go
Comment (iulianpascalau) [reply data/transaction/status.go:74]: agree, looks better with a new variable

File: data/transaction/status.go
Comment (iulianpascalau) [inline data/transaction/status.go:46]: missing nil checks for `uint64ByteSliceConverter` and `store`.

File: node/blockAPI/baseBlock.go
Comment (iulianpascalau) [inline node/blockAPI/baseBlock.go:107]: TODO still relevant?

Comment (valentincraciun99) [reply node/blockAPI/baseBlock.go:107]: yes

Comment (miiu96) [reply node/blockAPI/baseBlock.go:107]: Yes I will made a task for this TODO.

File: data/transaction/status.go
Comment (valentincraciun99) [reply data/transaction/status.go]: Done!

File: data/transaction/status.go
Comment (valentincraciun99) [reply data/transaction/status.go:47]: Done!

**Review by iulianpascalau — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by mateigraura — APPROVED** [review]

File: node/blockAPI/metaBlock.go
Comment (miiu96) [inline node/blockAPI/metaBlock.go:22]: create _statusComputer_ outside of `metaAPIBlockProcessor` and put in `APIBlockProcessorArg` as a member of this structure. `metaAPIBlockProcessor` should receiver `statusComputer` as an argument

File: node/blockAPI/shardBlock.go
Comment (miiu96) [inline node/blockAPI/shardBlock.go:21]: also here

**Review by mateigraura — DISMISSED** [review]

**Review by miiu96 — DISMISSED** [review]

File: data/transaction/status.go
Comment (valentincraciun99) [reply data/transaction/status.go]: Done!

File: data/transaction/status.go
Comment (valentincraciun99) [reply data/transaction/status.go]: Done!

File: data/transaction/status.go
Comment (valentincraciun99) [reply data/transaction/status.go]: Done!

File: node/blockAPI/metaBlock.go
Comment (valentincraciun99) [reply node/blockAPI/metaBlock.go]: Done!

File: node/blockAPI/shardBlock.go
Comment (valentincraciun99) [reply node/blockAPI/shardBlock.go]: Done!

**Review by iulianpascalau — APPROVED** [review]

**Review by iulianpascalau — CHANGES_REQUESTED** [review]
Will merge this after the soft-restart merge in development.

**Review by iulianpascalau — DISMISSED** [review]

File: node/blockAPI/baseBlock.go
Comment (iulianpascalau) [inline node/blockAPI/baseBlock.go:36]: please align unmarshalTx and txStatusComputer fields

Comment (valentincraciun99) [reply node/blockAPI/baseBlock.go]: Done!

**Review by iulianpascalau — APPROVED** [review]

**Review by mateigraura — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed

**Pull request merged**
